src/Facebook/InstantArticles/AMP/AMPArticle.php

```diff
@@ -97,15 +97,24 @@ private function __construct($instantArticle, $properties, $observer)
         $this->observer = $observer;
     }
 
-    public static function create($instantArticleString, $properties = array(), $observer = null)
+    /**
+     * Factory method to instantiate the AMPArticle converter.
+     * @param string|InstantArticle $instantArticle The instant article that will be parsed if informed as string.
+     * @param array() $properties the configuration for the AMP conversion. @see https://developers.facebook.com/docs/instant-articles/other-formats
+     * @param Observer $observer optional, will be created in case none informed. This is the hooking system for code level customizations.
+     */
+    public static function create($instantArticle, $properties = array(), $observer = null)
     {
-        libxml_use_internal_errors(true);
-        $document = new \DOMDocument();
-        $document->loadHTML($instantArticleString);
-        libxml_use_internal_errors(false);
+        // Treats if the informed content is string, parsing it into InstantArticle.
+        if (Type::is($instantArticle, Type::STRING)) {
+            libxml_use_internal_errors(true);
+            $document = new \DOMDocument('1.0');
+            $document->loadHTML($instantArticle);
+            libxml_use_internal_errors(false);
 
-        $parser = new Parser();
-        $instant_article = $parser->parse($document);
+            $parser = new Parser();
+            $instantArticle = $parser->parse($document);
+        }
 
```

Comment:
Should have a test here to see if $instantArticle is really an instance of an IA.
If the user passes something else than string but not an IA, it will break